mlrun/runtimes/local.py

```diff
@@ -275,11 +290,17 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
             try:
                 fn = self._get_handler(handler, context)
                 global_context.set(context)
+                # Running tracking services pre run to detect if some of them should be used:
+                trackers_manager = self._get_trackers_manager()
+                trackers_manager.pre_run(context)
                 sout, serr = exec_from_params(fn, runobj, context)
                 log_std(
                     self._db_conn, runobj, sout, serr, skip=self.is_child, show=False
                 )
+                # If trackers where used, this is where we log all data collected to MLRun
+                context = trackers_manager.post_run(context)
```

Comment:
imo code dup. I wonder if we want to wrap that functionality all together

Reply:
My take on it: 

The whole tracker manager thingy is simply a callback mechanism. I asked Orit if we plan on adding callbacks to MLRun functions (which this can be a `MLFlowCallback`) but she said not yet so this was added to wrap the execution. 

In a complete solution there should be a callbacks handler instance with the user's added callbacks that will call there `on_run_begin` and `on_run_end` (and it will have `on_run_error` and so on). For now I think this is good enough as removing this duplications will require adjusting the whole code here (which should be done regardless but not in this PR)